src/serde.rs

```diff
@@ -675,17 +678,20 @@ mod test {
         ];
         for &raw in data.iter() {
             let value = Decimal::from_str(raw).unwrap();
-            let encoded = serialize(&value).unwrap();
-            let decoded: Decimal = deserialize(&encoded[..]).unwrap();
+            let encoded = encode_to_vec(&value, config::standard()).unwrap();
+            let decoded: Decimal = decode_from_slice(&encoded[..], config::standard()).unwrap().0;
             assert_eq!(value, decoded);
-            assert_eq!(8usize + raw.len(), encoded.len());
+            assert_eq!(1usize + raw.len(), encoded.len());
```

Comment:
Not sure this is correct. I check the str serialization path is running, but I'm fuzzy with the internal representation. Not sure why it would be smaller tbh